joblib/test/test_pickle.py

```diff
@@ -0,0 +1,21 @@
+"""
+Test that pickle and dill output the same bytes
+"""
+import pickle
+
+import nose
+
+from joblib.test.common import dill, with_dill, with_py3
+
+
+def global_func():
+    pass
+
+
+@with_py3
+@with_dill
+def test_dumps_eq():
+    # Test that dill and pickle produce the same results
```

Comment:
@lesteve Good point, I'll dig deeper and find out if dill supports the python 2 protocol or not.
@ogrisel That would work but I'd really like to use the hashing functionality, I personally need this for caching objects whose classes use decorators.